src/dtgtk/thumbtable.c

```diff
@@ -1591,7 +1593,7 @@ static void _dt_mouse_over_image_callback(gpointer instance, gpointer user_data)
           // right border
           b = TRUE;
           if(table->mode != DT_THUMBTABLE_MODE_FILMSTRIP
-             && pos < g_list_length(table->list) - 1
+             && pos < table_len - 1
```

Comment:
or we can just test if `g_list_next(l)` is valid... but as I'm not sure how to remove the next call to `g_list_length` maybe it's more clear to write it like that...